research/query_service/ir/compiler/src/main/java/com/alibaba/graphscope/common/IrPlan.java

```diff
@@ -46,9 +46,12 @@
 // represent ir plan as a chain of operators
 public class IrPlan implements Closeable {
     private static IrCoreLibrary irCoreLib = IrCoreLibrary.INSTANCE;
-    private static String PLAN_JSON_FILE = "plan.json";
+    // write to '/tmp' which is accessible from common users
+    private static String JSON_PLAN_DIR = "/tmp";
```

Comment:
Can we obtain temporary fold from some environment variables. For example, Mac's temporary fold is not at "/tmp"

Reply:
I think `/tmp` is ok in macOS, which is a link to `/private/tmp`, Or maybe you can get the tmp directory by `System.getProperty('java.io.tmpdir')`

> https://kodejava.org/how-do-i-get-operating-system-temporary-directory-folder/

```java
package org.kodejava.lang;

public class TempDirExample {
    public static void main(String[] args) {
        // This is the property name for accessing OS temporary directory
        // or folder.
        String property = "java.io.tmpdir";

        // Get the temporary directory and print it.
        String tempDir = System.getProperty(property);
        System.out.println("OS temporary directory is " + tempDir);
    }
}
```

Reply:
done